common/display/displayplanemanager.cpp

```diff
@@ -458,6 +460,39 @@ void DisplayPlaneManager::ValidateCachedLayers(
   *render_layers = needs_gpu_composition;
 }
 
+bool DisplayPlaneManager::ValidateSurfaceDamage(
+    const std::vector<OverlayLayer> *layers,
+    const std::vector<OverlayLayer> *previous_layers) {
+  auto layer_begin = layers->begin();
+  auto layer_end = layers->end();
+  bool validate = true;
+  size_t num_layers = layers->size();
+
+  for (size_t i = 0; i <= num_layers; i++) {
+    const HwcRegion &current_surface_damage = layers->at(i).GetSurfaceDamage();
+
+    if (current_surface_damage.kNumRects == 1) {
+      const HwcRect<int> *rect = current_surface_damage.kRects;
+      if ((rect->top == 0) && (rect->bottom == 0) && (rect->left == 0) &&
+          (rect->right == 0)) {
+        const HwcRect<int> &previous = previous_layers->at(i).GetDisplayFrame();
+        const HwcRect<int> &current = layers->at(i).GetDisplayFrame();
+        if ((previous.left != current.left) || (previous.top != current.top)) {
+          validate = true;
+          break;
+        }
+        validate = false;
+        continue;
+      }
+    } else if (current_surface_damage.kNumRects == 0) {
+      validate = true;
```

Comment:
This patch is the basic patch which will avoid the complete compose and flip if the all the layers are not changed. In the next final set of patches with all the optimizations including per layer damage in that scenario am planning to reuse the ValidateCachedLayer function as such and will consider your review comments also. 
Also one more think in the ValidateCachedLayer we are allocating the buffer not reusing the previous layer buffers. 

For the last comment we cant replace the left and right checks there are teh scenarios where the buffer content not changed but the src x and y co-ordinates changed in that case we need to update the src co-ordinates accordingly